subprojects/core/src/test/groovy/org/gradle/api/internal/plugins/DefaultPluginContainerTest.groovy

```diff
@@ -103,6 +103,27 @@ class DefaultPluginContainerTest extends Specification {
         !container.hasPlugin(plugin2Class)
     }
 
+    def "offers plugin management via injectable plugin type "() {
+        when:
+        def basePluginClass = classLoader.parseClass("""
+        import org.gradle.api.Plugin
+        import org.gradle.api.Project
+        class TestPlugin1 implements Plugin<Project> {
+          void apply(Project project) {}
+          @javax.inject.Inject
+          public org.gradle.internal.reflect.Instantiator getter() {
+            throw new NullPointerException();
```

Comment:
We typically throw `UnsupportedOperationException` from these getters. Another option is to use an abstract getter.

Reply:
Also, I think we should have this as a proper getter, i.e. `getInstantiator()`.

Reply:
Do we even inject things that are not true JavaBean getters? 🤔

Reply:
Agreed, `getter` was really a stupid name but it seems it was good enough to trigger the injection behaviour